File: CA269/PhoneNumbers.java
import java.util.Scanner;
import java.util.Map;
import java.util.HashMap;

public class PhoneNumbers
{
    public static void main(String [] args)
    {
    	Scanner in = new Scanner(System.in);
    	Map<String, String> numbers = new HashMap <String, String>();
    	System.out.println("Enter a name and number, or a name and ? to query (!! to exit)");
    	String s = in.nextLine();
    	while(! s.equals("!!"))
    	{
    		if(s.substring(s.length() - 2, s.length()).equals(" ?"))
    		{
    			String [] name = s.split(" ");
    			if(numbers.containsKey(name[0]))
    				System.out.println(name[0] + " has number " + numbers.get(name[0]));
    			else
    				System.out.println("Sorry, there is no " + name[0]);
    		}
    		else
    		{
    			String [] namenum = s.split(" ");
    			numbers.put(namenum[0], namenum[1]);
    		}
    		s = in.nextLine();
    	}
    	System.out.println("Bye");
    }
}

File: CA268_Programming3/PhoneNumbers.py
def main():
	print("Enter a name and number, or a name and ? to query (!! to exit)")
	inp = input()
	d = {}
	while inp != "!!":
		inp = inp.strip().split()
		if inp[1] == "?":
			if inp[0] in d:
				print(str(inp[0]) + " has number " + str(d[inp[0]]))
			else:
				print("Sorry, there is no " + str(inp[0]))
		else:
			d[inp[0]] = int(inp[1])
		inp = input()
	print("Bye")

if __name__ == '__main__':
	main()
